docs/lesson-plans/week-5.md

```diff
@@ -0,0 +1,639 @@
+
+#### Parameterized Tests
+
+As said previously, you can also run multiple tests using different input parameters by using the `@ParameterizedTest` 
+annotation. 
+
+Inputs to your method can be provided using another annotation, which can consist of other annotations.
+Three of the most basic ones are `@NullSource`, `@EmptySource` (provides an empty String, List, Set, Map, or any 
+array type), and the combination `@NullAndEmptySource`. Those can be combined with `@ValueSource` (or you can use them 
+alone if that suits your needs better) to test with a simple set of values.
+
+The `@ValueSource` annotation allows you to provide a list of values of any atomic type (short, byte, int, long, float,
+double, char, boolean), as well as String and `Class` types. 
+
+>Note: the input to the annotation must be the pluralized name of the type you're inputting. As an example, a list of 
+>integer inputs would be `@ValueSource(ints = {1, 2, 3, ...})`, and booleans would 
+>be `@ValueSource(booleans = {true, false})`.
+
+The `@EnumSource` conveniently provides all sub-types of a provided enum class.
+
+If none of those suit your needs, you can also use `@MethodSource`, `@CsvSource`, `@CsvFileSource`, and 
+`@ArgumentSource`.
+
+You can see all of the sources of parameterized arguments [on the JUnit 
+docs](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources).
+
+Here is an example of a parameterized test using integer inputs:
+
+```java
+@ParameterizedTest
+@ValueSource(ints = {1, 2, 3})
+@NullSource
+// Notice how we're using the wrapped integer class "Integer" since we're inputting a null source.
+public void testParameterizedTest(Integer value) { 
+    if (value == null) {
+        assertNull(value);
+    }
+    else {
+        assertTrue(value > 0 && value < 4);
+    }
+}
+```
+
+##### Externals Class
+
+Sometimes you end up testing things where you need to supply your own value (especially if the method providing a value 
+is static). For example, when getting a `Router` object in `PostsRouter`. If we have our own version of a `Router` 
+object which we want to supply for testing purposes, we need a way to substitute that in.
+
+An `Externals` class (sometimes abbreviated to `externs`) can allow us to do that. We can define a 'nested'
+`public static class Externals` as a nested class in `PostsRouter`, which has a method 
+`public Router getRouter(Vertx vertx)`, and that will allow us to do so. In the default `Externals` 
+(the one in `PostsRouter`),  we'll start out just having that single `getRouter` method. Then, in our test class,
+we'll extend that `Externals` class with a new `TestExternals` class, and override the `getRouter` method to have it 
+return our custom router object. This can be applied in many different ways for times where you need to override 
+behavior in dependencies needed for testing.
+
```

Comment:
Make sure all of these side notes render in mkdocs as different sections, I'm pretty sure they won't as they are now